File: src/main/java/com/truveta/opentoken/attributes/validation/RegexValidator.java
/**
 * Copyright (c) Truveta. All rights reserved.
 */
package com.truveta.opentoken.attributes.validation;

import java.util.regex.Pattern;
import javax.validation.constraints.NotNull;

import lombok.Getter;
import lombok.Setter;

/**
 * A Validator that is designed for validating with regex expressions.
 */
@Getter
@Setter
public final class RegexValidator implements AttributeValidator {

    @NotNull
    private String attributeName;
    private final Pattern compiledPattern;

    public RegexValidator(@NotNull String attributeName, @NotNull String pattern) {
        this.attributeName = attributeName;
        this.compiledPattern = Pattern.compile(pattern);
    }

    /**
     * Validates that the value matches the regex pattern.
     */
    @Override
    public boolean eval(String name, String value) {
        return (!name.equals(attributeName)) ||
                (value != null && compiledPattern.matcher(value).matches());
    }
}

File: lib/nodejs/opentoken/src/attributes/validation/RegexValidator.ts
/**
 * Copyright (c) Truveta. All rights reserved.
 */

import { AttributeValidator } from './AttributeValidator';

/**
 * A validator that validates a value against a regular expression.
 */
export class RegexValidator implements AttributeValidator {
  private pattern: RegExp;

  /**
   * Constructs a new RegexValidator.
   *
   * @param pattern - The regular expression pattern to validate against.
   */
  constructor(pattern: string | RegExp) {
    this.pattern = typeof pattern === 'string' ? new RegExp(pattern) : pattern;
  }

  /**
   * Validates the given value against the regular expression.
   *
   * @param value - The value to validate.
   * @returns true if the value matches the pattern; false otherwise.
   */
  validate(value: string): boolean {
    if (!value) {
      return false;
    }
    return this.pattern.test(value);
  }
}
